src/librustc_target/spec/aarch64_apple_darwin.rs

```diff
@@ -0,0 +1,32 @@
+use crate::spec::{LinkerFlavor, Target, TargetOptions, TargetResult};
+
+pub fn target() -> TargetResult {
+    let mut base = super::apple_base::opts();
+    base.cpu = "apple-a12".to_string();
+    base.max_atomic_width = Some(128);
+    base.eliminate_frame_pointer = false;
+    base.pre_link_args.insert(LinkerFlavor::Gcc, vec!["-arch".to_string(), "arm64".to_string()]);
+
+    base.link_env_remove.extend(super::apple_base::macos_link_env_remove());
+    base.stack_probes = true;
+
+    // Clang automatically chooses a more specific target based on
+    // MACOSX_DEPLOYMENT_TARGET.  To enable cross-language LTO to work
+    // correctly, we do too.
+    let arch = "aarch64";
+    let llvm_target = super::apple_base::macos_llvm_target(&arch);
+
+    Ok(Target {
+        llvm_target,
+        target_endian: "little".to_string(),
+        target_pointer_width: "64".to_string(),
+        target_c_int_width: "32".to_string(),
```

Comment:
```
% cat size.c
#include <stdio.h>

int main() {
  printf("%zu\n", sizeof(int));
  return 0;
}

% cc size.c

% file a.out
a.out: Mach-O 64-bit executable arm64

% ./a.out
4
```

Reply:
You might also want `printf("%zu %zu\n", sizeof(int*), sizeof(int (*)()));`. Not that I expect it to be anything other than 8 bytes.